### PR TITLE
add `oomd` plugin

### PR DESCRIPTION
**Pull request by jdockerty**
Plugin for showing `OOMKilled` pods

E.g.

```
kubectl oomlie -n oomkilled

POD                        CONTAINER        REQUEST     LIMIT     TERMINATION TIME
my-app-5bcbcdf97-722jp     infoapp          1G          8G        2022-11-07 13:03:49 +0000 GMT
my-app-5bcbcdf97-7j5rd     infoapp          1G          8G        2022-11-07 14:35:34 +0000 GMT
my-app-5bcbcdf97-k8g8g     infoapp          1G          8G        2022-11-07 14:35:02 +0000 GMT
my-app-5bcbcdf97-mf65j     infoapp          1G          8G        2022-11-07 14:34:57 +0000 GMT

```

### CODE REVIEW

**Comment by ahmetb** [comment]
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

Thanks for submitting your kubectl plugin to Krew!
One of the krew-index maintainers will review it soon. Note that the reviews for new plugin submissions may take a few days.

In the meanwhile, here are a few tips to make your plugin manifest better:
* Make sure your plugin follows the [best practices](https://krew.sigs.k8s.io/docs/developer-guide/develop/best-practices/).
* Eliminate redundant wording form `shortDescription` (it should be max 50 characters).
* Try to word wrap your `description` to 80-character lines (no usage examples, please).

Thanks for your patience!
/kind new-plugin

File: plugins/oomlie.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        arch: amd64
+    uri: "https://github.com/jdockerty/kubectl-oomlie/releases/download/v0.0.1/oomlie_windows_amd64.zip"
+    sha256: "32c3a1bc1091385b7bab0c85888e93805a3352d9fc548f9542fb59fb87368ecf"
+    files:
+    - from: "/oomlie.exe"
+      to: "."
+    - from: LICENSE
+      to: "."
+    bin: "oomlie.exe"
+  shortDescription: Show pods which have recently been OOMKilled
```
Comment (ahmetb) [inline plugins/oomlie.yaml:44]: > Show recently OOMKilled Pods

**Review by ahmetb — COMMENTED** [review]
Thanks for the submission. I think the name's a bit hard to find/memorize, but it's a personal choice so I'll let you decide.

Do you think showing memory limit/requests would be useful in the output?

File: plugins/oomlie.yaml (hunk 1 above)
Comment (jdockerty) [reply plugins/oomlie.yaml:44]: Much more concise, I like it :rocket: 

Changing this now :blush:

**Comment by jdockerty** [comment]
@ahmetb Much appreciated!

> I think the name's a bit hard to find/memorize, but it's a personal choice so I'll let you decide.

I'm open to a suggestion if you have any :blush: I went back and forth with various things, although it always came out as though it's forcing a pod to OOM or something along those lines.

The `oomlie` term seemed to get the point across and sounded funny/snappy, so I went with that. Although happy to change it though if you have any suggestions.

> Do you think showing memory limit/requests would be useful in the output?

This is actually a really good idea and now I'm disappointed that I didn't think of that myself :smile: 

I was keeping the plugin simple for a `v0.0.1` iteration, although I could certainly add it if you think that it is something which is required for a merge to the index.

Either way I'll add it in, it'll be a `v0.1.1` thing if it is merged beforehand and I'll let the workflow open a new release PR :blush:

**Comment by ahmetb** [comment]
Alt names could be: oomd, oomed, oomkilled ¯\_(ツ)_/¯

**Comment by ahmetb** [comment]
You could also make this a more generic plugin that looks as other termination conditions, not just oom maybe. We have a plugin that finds restarts/crashlooping pods as well.

**Comment by jdockerty** [comment]
I'll update this name to `oomd`, since it's easier to find and remember that way :blush: 

> You could also make this a more generic plugin that looks as other termination conditions, not just oom maybe. We have a plugin that finds restarts/crashlooping pods as well.

I'll leave this as specifically `OOMKilled` pods if that is okay with you? It fulfils a personal/work issue for me and it's also nice to share with others :blush: I'll put something together for the memory limits/requests too and push that up.

**Pull request merged**